### ISSUE TITLE
JSCS does not fix disallowTrailingWhitespace errors on Windows

### ISSUE THREAD

**Issue by mpawlow** [jscs-dev/node-jscs#1939]
*Environment*
---

* Windows 7 (64-bit)
* Source file being validated is in UTF-8 Unix format (i.e. LFs only) 
* Source file contains a line with trailing whitespace.
* JSCS version: 2.4.0 

*Steps To Reproduce*
---

(1) Validate source file using JSCS
```
jscs .\lib\utils\http-util.js
```
* Note: Error is reported as expected
```
disallowTrailingWhitespace: Illegal trailing whitespace at .\lib\utils\http-util.js
```

(2) Fix source file Using JSCS
```
jscs .\lib\utils\http-util.js --fix
```
* Note: Appears to have worked. No output / errors.
* Note: The source file is modified (i.e. file opened editor indicates file has changed and reloads it)

(3) Validate source file using JSCS again
```
jscs .\lib\utils\http-util.js
```
* BUG: Error is not fixed
```
disallowTrailingWhitespace: Illegal trailing whitespace at .\lib\utils\http-util.js
```

*Important Notes*
---

* Noticed that other whitespace issues ARE fixed - just not trailing whitespace.
* This is NOT a problem on Unix (verified on Mac)

*JSCS Configuration*
---

```
{
	"verbose": true,
	"esnext": false,

	"validateIndentation": "\t",
	"validateQuoteMarks": true,
	"requireCurlyBraces": [
		"if",
		"else",
		"for",
		"while",
		"do",
		"try",
		"catch"
	],
	"requireSpaceAfterKeywords": [
		"if",
		"else",
		"for",
		"while",
		"do",
		"switch",
		"case",
		"return",
		"try",
		"catch",
		"typeof"
	],
	"requireSpaceBeforeBlockStatements": true,
	"requireParenthesesAroundIIFE": true,
	"requireSpacesInConditionalExpression": true,
	"disallowSpacesInNamedFunctionExpression": {
		"beforeOpeningRoundBrace": true
	},
	"disallowSpacesInFunctionDeclaration": {
		"beforeOpeningRoundBrace": true
	},
	"disallowSpacesInFunctionExpression": {
		"beforeOpeningRoundBrace": true
	},
	"disallowSpacesInAnonymousFunctionExpression": {
		"beforeOpeningRoundBrace": true
	},
	"requireSpaceBetweenArguments": true,
	"requireMultipleVarDecl": false,
	"requireVarDeclFirst": false,
	"requireBlocksOnNewline": true,
	"disallowEmptyBlocks": true,
	"disallowSpacesInsideArrayBrackets": true,
	"disallowSpacesInsideParentheses": true,
	"disallowDanglingUnderscores": false,
	"requireCommaBeforeLineBreak": true,
	"disallowSpaceAfterPrefixUnaryOperators": true,
	"disallowSpaceBeforePostfixUnaryOperators": true,
	"disallowSpaceBeforeBinaryOperators": [
		","
	],
	"requireSpacesInForStatement": true,
	"requireSpacesInAnonymousFunctionExpression": {
		"beforeOpeningCurlyBrace": true
	},
	"requireSpaceBeforeBinaryOperators": true,
	"requireSpaceAfterBinaryOperators": true,
	"disallowKeywords": [
		"with"
	],
	"disallowMixedSpacesAndTabs": true,
	"disallowTrailingWhitespace": true,
	"disallowTrailingComma": true,
	"disallowKeywordsOnNewLine": [
		"else"
	],
	"requireLineFeedAtFileEnd": true,
	"requireCapitalizedConstructors": true,
	"requireDotNotation": true,
	"disallowNewlineBeforeBlockStatements": true,
	"disallowMultipleLineStrings": true,
	"requireSpaceBeforeObjectValues": true,
	"disallowSpacesInCallExpression": true,
	"disallowSpaceAfterObjectKeys": true,
	"disallowYodaConditions": true,
	"disallowMultipleLineBreaks": true,
	"disallowMultipleVarDecl": true,
	"disallowSpaceBeforeComma": true,
	"disallowSpaceBeforeSemicolon": true,
	"requireCamelCaseOrUpperCaseIdentifiers": true,
	"requirePaddingNewLinesBeforeLineComments": {
		"allExcept": "firstAfterCurly"
	},
	"requirePaddingNewLinesAfterBlocks": true,
	"requireSemicolons": true,
	"safeContextKeyword": "_this",
	"validateLineBreaks": "LF"
}
```

**Comment by lukeapage** [comment]
I found it works, but you have to run it multiple times - it took me 4 runs to remove all trailing space...

**Comment by lukeapage** [comment]
may be fixed by one of #1978, #1979, #1980

Needs a snippet of code that has the issue in...

**Comment by markelog** [comment]
At this point only major and CST related bugs will be fixed.
